### PR TITLE
[MKL-DNN] Conv grad and Batch Norm grad NHWC support

### PR DESCRIPTION
**Pull request by jczaja**
This PR is introducing support to NHWC Paddle layout for conv and batch norm grad ops as discussed #20964.

### CODE REVIEW

**Review by grygielski — APPROVED** [review]
LGTM

**Comment by bingyanghuang** [comment]
@luotao1 Please start a review.

File: paddle/fluid/platform/mkldnn_helper.h [hunk 1]
```diff
@@ -74,6 +74,11 @@ tf_pd<Type> MKLDNNBwdPrimitiveDesc(const Engine& e, const Primitive& p,
 inline void MatchShapeToLayout(framework::Tensor* tensor_in,
                                framework::DataLayout from,
                                framework::DataLayout to) {
+  // Shape changing makes sense for 3+ dims Tensors
+  if (tensor_in->dims().size() < 3) {
+    return;
+  }
+
```
Comment (LeoZhao-BJ) [inline paddle/fluid/platform/mkldnn_helper.h:81]: what's expected behavior if dims <3 ? nothing to do?

**Review by LeoZhao-BJ — COMMENTED** [review]
LGTM

File: paddle/fluid/platform/mkldnn_helper.h (hunk 1 above)
Comment (grygielski) [reply paddle/fluid/platform/mkldnn_helper.h:81]: In these data layouts, channel dimension is either on 2nd position: nChw or at last nhwC, so for dim==2 these layouts are the same and nothing should be done. Similarly for dim==1 when you have just one possible combination.

Comment (luotao1) [reply paddle/fluid/platform/mkldnn_helper.h:81]: @grygielski Could you add the above comments in the codes in the next PR?

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**